libtcmu.c

```diff
@@ -913,6 +912,7 @@ static void remove_device(struct tcmulib_context *ctx,
 	if (ret < 0) {
 		tcmu_err("could not cleanup mailbox lock %s: %d\n", dev_name, errno);
 	}
+	dev->handler->removed(dev);
```

Comment:
Wouldn't it be a better fix if `free(dev)` is called in `remove_device()` rather than in the thread cleanup handler? The thread cleanup handler would then just call `r_handler->close()`.

The point of moving `dev->handler->removed` before closing `dev->fd` is to cleanup any users of `dev->fd` before closing`fd`.